src/Sentry/TransactionTracer.cs

```diff
@@ -198,9 +200,15 @@ public TransactionTracer(IHub hub, string name, string operation, TransactionNam
     /// <summary>
     /// Initializes an instance of <see cref="TransactionTracer"/>.
     /// </summary>
-    public TransactionTracer(IHub hub, ITransactionContext context)
+    public TransactionTracer(IHub hub, ITransactionContext context) : this (hub, context, null) {}
+
+    /// <summary>
+    /// Initializes an instance of <see cref="TransactionTracer"/>.
+    /// </summary>
+    public TransactionTracer(IHub hub, ITransactionContext context, TimeSpan? idleTimeout)
```

Comment:
Can we read the options through `hub` and avoid adding the overload?

It's internal for that reason: https://github.com/getsentry/sentry-dotnet/blob/69a92b54fb9254aca451bf9ff038154f053f6b69/src/Sentry/Internal/Hub.cs#L28

Reply:
Yes, that could be done.
`IHub` has more than one implementation but only `Hub` has the internal `SentryOptions`. Is the transaction tracer guaranteed to be invoked by `Hub` and be able to cast to `Hub` to access `SentryOptions` or not? And if not, should the failover be idle timeout as null? I can see that working.

Reply:
Implementations of `IHub` are

1. [`Hub`](https://github.com/getsentry/sentry-dotnet/blob/af5aac5a3d0e5faddda57d68a6ff7d4f169aab88/src/Sentry/Internal/Hub.cs#L7C5-L7C5)
2. [`DisabledHub`](https://github.com/getsentry/sentry-dotnet/blob/af5aac5a3d0e5faddda57d68a6ff7d4f169aab88/src/Sentry/Extensibility/DisabledHub.cs#L6C14-L6C25)
3. [`HubAdapter`](https://github.com/getsentry/sentry-dotnet/blob/af5aac5a3d0e5faddda57d68a6ff7d4f169aab88/src/Sentry/Extensibility/HubAdapter.cs#L15)

Only `Hub` has the SentryOptions. The other two implementations are used for testing.
I've decided to cast `IHub` to `Hub explicitly` and, if not `null`, access the options. This should be a fair assumption.

Reply:
Addressed